### PR TITLE
feat: add served_model_name argument for the infinity_server

### LINKED ISSUE #178: model name is not consistent across endpoints
### Feature request

Add a `--served-model-name` option to control the model name.

### Motivation

I ran:

```sh
docker run -p 8080:8080 michaelf34/infinity:latest --model-name-or-path BAAI/bge-m3 --port 8080
```

Query the models endpoint:

```sh
$ curl -s http://0.0.0.0:8080/models | jq
{
  "data": [
    {
      "id": "BAAI/bge-m3",
      "stats": {
        "queue_fraction": 0,
        "queue_absolute": 0,
        "results_pending": 0,
        "batch_size": 32
      },
      "object": "model",
      "owned_by": "infinity",
      "created": 1711612054,
      "backend": "torch"
    }
  ],
  "object": "list"
}
```

Query the embeddeings endpoint:

```sh
$ curl -s -X 'POST'   'http://0.0.0.0:8080/embeddings'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{
  "input": [
    "string"
  ]}' | jq | grep model
  "model": "BAAIbge-m3",
```

Via embeddings endpoint the model is `BAAIbge-m3` for the model endpoint it is `BAAI/bge-m3`, Somehow would be cool to control the name.

[vLLM is doing this e.g. with the following options](https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html#command-line-arguments-for-the-server): 

- `--served-model-name`:  The model name used in the API. If not specified, the model name will be the same as the huggingface name.
- `--model`:  name or path of the huggingface model to use

### Your contribution

I can create a PR  for this

### PR DESCRIPTION
**Pull request by bufferoverflow**
Closes #178

### CODE REVIEW

File: libs/infinity_emb/infinity_emb/infinity_server.py [hunk 1]
```diff
@@ -53,7 +53,12 @@ def create_server(
     instrumentator = Instrumentator().instrument(app)
     app.add_exception_handler(errors.OpenAIException, errors.openai_exception_handler)
 
-    model_name_response_name = "".join(engine_args.model_name_or_path.split("/")[-2:])
+    if engine_args.served_model_name:
```
Comment (michaelfeil) [inline libs/infinity_emb/infinity_emb/infinity_server.py:56]: Can you make this a one liner?

```python
model_name_response_name =  (
      engine_args.served_model_name or \
      "/".join( # added /
            engine_args.model_name_or_path.split("/")[-2:]
        )
)
```

File: libs/infinity_emb/infinity_emb/infinity_server.py [hunk 2]
```diff
@@ -107,7 +112,7 @@ async def _models():
         return dict(
             data=[
                 dict(
-                    id=engine_args.model_name_or_path,
+                    id=model_name_response_name,
```
Comment (michaelfeil) [inline libs/infinity_emb/infinity_emb/infinity_server.py:115]: Also missing this engine_args.served_model_name in `/embeddings` reponse

File: libs/infinity_emb/infinity_emb/infinity_server.py (hunk 1 above)
Comment (bufferoverflow) [reply libs/infinity_emb/infinity_emb/infinity_server.py]: of course, done

File: libs/infinity_emb/infinity_emb/infinity_server.py (hunk 2 above)
Comment (bufferoverflow) [reply libs/infinity_emb/infinity_emb/infinity_server.py:112]: Isn't that done [here](https://github.com/michaelfeil/infinity/blob/main/libs/infinity_emb/infinity_emb/infinity_server.py#L154) already ?

Comment (michaelfeil) [reply libs/infinity_emb/infinity_emb/infinity_server.py:112]: Thats right - my bad!

**Comment by michaelfeil** [comment]
Sorry for the rebase, did not expect you to move that fast with the contrib!

**Comment by bufferoverflow** [comment]
> Sorry for the rebase, did not expect you to move that fast with the contrib!

all good, happy to use your software!

**Review by michaelfeil — APPROVED** [review]

**Pull request merged**